spec/models/application_template_spec.rb

```diff
@@ -0,0 +1,79 @@
+require 'rails_helper'
+
+describe ApplicationTemplate do
+  describe 'create_draft' do
+    before :each do
+      @application_template = create :application_template
+      create :question, application_template: @application_template
+      create :question, application_template: @application_template
+      @user = create :user
+    end
+    let :call do
+      @application_template.create_draft @user
+    end
+    context 'pre-existing draft belonging to user' do
+      before :each do
+        create :application_draft,
+               application_template: @application_template,
+               user: @user
+      end
+      it 'returns false' do
+        expect(call).to eql false
+      end
+    end
+    context 'no pre-existing draft' do
+      it 'creates an application template draft' do
+        expect { call }.to change { ApplicationDraft.count }.by 1
+      end
+      it 'returns the draft' do
+        expect(call).to be_a ApplicationDraft
+      end
+      it 'sets the user of the draft to the user argument' do
+        expect(call.user). to eql @user
+      end
+      it 'sets the application template of the draft to the current one' do
+        expect(call.application_template).to eql @application_template
+      end
+      it 'adds the questions of the application template to the draft' do
+        # there are no questions in the draft, so this fails.
+        expect(call.questions.size).to eql @application_template.questions.size
+      end
+    end
+  end
+  describe 'draft_belonging_to' do
+    before :each do
+      @user = create :user
+      other_user = create :user
+      @application_template = create :application_template
+      @draft = create :application_draft,
+                      application_template: @application_template,
+                      user: @user
+      # other draft - why do we need this?
```

Comment:
Good question - so that we can be sure the method is returning what we expect it to, and not what we don't expect it to. The only way we can do this is by making data we expect it not to return. Say, for example, if the method just returned `ApplicationDraft.all`.